### PR TITLE
Autodetect OVN octavia provider

### PR DESCRIPTION
**Pull request by luis5tb**
This makes the CNO to detect if ovn provider is configured in
octavia, and if that so configures kuryr to use it. Thanks to this
the need for a VM per svc is removed and loadbalancing is done by
using the ovn distributed loadbalancer (aka ovn flows)

### CODE REVIEW

File: pkg/platform/openstack/kuryr_bootstrap.go [hunk 1]
```diff
@@ -1077,6 +1081,31 @@ func BootstrapKuryr(conf *operv1.NetworkSpec, kubeClient client.Client) (*bootst
...
+		page, err := providers.List(lbClient).AllPages()
+		if err != nil {
+			log.Print("failed to get lbs provider list, using default octavia provider")
+		} else {
+			providers, err := providers.ExtractProviders(page)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to extract providers list")
+			}
+			for _, provider := range providers {
+				if provider.Name == OVNProvider {
```
Comment (dulek) [inline pkg/platform/openstack/kuryr_bootstrap.go]: Is this a standardized name or can admin name providers however he likes? I mean - is it like flavors?

File: pkg/platform/openstack/kuryr_bootstrap.go [hunk 2]
```diff
@@ -624,8 +628,8 @@ func ensureOpenStackLbMonitor(client *gophercloud.ServiceClient, name, poolId st
 			PoolID:     poolId,
 			Type:       monitors.TypeTCP,
 			MaxRetries: 3,
-			Delay:      10,
-			Timeout:    10,
+			Delay:      5,
+			Timeout:    4,
```
Comment (dulek) [inline pkg/platform/openstack/kuryr_bootstrap.go]: Why those changes?

File: pkg/platform/openstack/kuryr_bootstrap.go [hunk 3]
```diff
@@ -22,6 +22,7 @@ import (
 	"github.com/gophercloud/gophercloud/openstack/loadbalancer/v2/loadbalancers"
 	"github.com/gophercloud/gophercloud/openstack/loadbalancer/v2/monitors"
 	"github.com/gophercloud/gophercloud/openstack/loadbalancer/v2/pools"
+	"github.com/gophercloud/gophercloud/openstack/loadbalancer/v2/providers"
```
Comment (dulek) [inline pkg/platform/openstack/kuryr_bootstrap.go]: I think you need to regenerate `dep` after that.

File: pkg/platform/openstack/kuryr_bootstrap.go (hunk 1 above)
Comment (luis5tb) [reply pkg/platform/openstack/kuryr_bootstrap.go]: That is static, and seems it must be 'ovn' for the ovn-octavia driver

File: pkg/platform/openstack/kuryr_bootstrap.go (hunk 3 above)
Comment (luis5tb) [reply pkg/platform/openstack/kuryr_bootstrap.go]: yep! I'll update now that gophercloud side got merged

**Pull request merged**